### PR TITLE
SONARJAVA-2710 Rule S3077: Non-primitive fields should not be "volatile"

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java [hunk 1]
```diff
@@ -54,12 +54,12 @@ public void visitNode(Tree tree) {
...
     Type varType = variableTree.type().symbolType();
-    if(varType.isArray()) {
+    if (varType.isArray()) {
       String atomicType = "Reference";
       Type elementType = ((Type.ArrayType) varType).elementType();
-      if(elementType.isPrimitive(Type.Primitives.LONG)) {
+      if (elementType.isPrimitive(Type.Primitives.LONG)) {
         atomicType = "Long";
-      } else if(elementType.isPrimitive(Type.Primitives.INT)) {
+      } else if (elementType.isPrimitive(Type.Primitives.INT)) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java]: ok i'll try to pay attention to this...

Comment (mpaladin) [reply java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java]: 😉

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3077_java.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "constantCost": "20min"
+  },
+  "tags": [
+    "multi-threading",
+    "cert"
+  ],
+  "standards": [
+    "CERT"
+  ],
+  "defaultSeverity": "Major",
```
Comment (mpaladin) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3077_java.json]: @benzonico any thought about being major bug?

Comment (benzonico) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3077_java.json]: The "bug" might be arguable given the amount of issues in ruling (and in jdk/guava especially)

**Pull request merged**